docs/reference/creating_a_provider_class.rst

```diff
@@ -253,29 +253,35 @@ Register the Class with the Service Container
 If you use the tag ``sonata.media.provider``, the provider service will be
 added to the provider pool.
 
-.. code-block:: xml
-
-    <!-- config/services.xml -->
-
-    <service id="sonata.media.provider.vimeo" class="Sonata\MediaBundle\Provider\VimeoProvider">
-        <argument>sonata.media.provider.vimeo</argument>
-        <argument type="service" id="sonata.media.filesystem.local"/>
-        <argument type="service" id="sonata.media.cdn.server"/>
-        <argument type="service" id="sonata.media.generator.default"/>
-        <argument type="service" id="sonata.media.thumbnail.format"/>
-        <argument type="service" id="sonata.media.buzz.browser"/>
-        <argument type="service" id="sonata.media.metadata.proxy"/>
-        <call method="setTemplates">
-            <argument type="collection">
-                <argument key='helper_thumbnail'>@SonataMedia/Provider/thumbnail.html.twig</argument>
-                <argument key='helper_view'>@SonataMedia/Provider/view_vimeo.html.twig</argument>
-            </argument>
-        </call>
-        <call method="setResizer">
-            <argument type="service" id="sonata.media.resizer.simple"/>
-        </call>
-        <tag name="sonata.media.provider"/>
-    </service>
+.. code-block:: yaml
+
+    # config/services.yaml
+
+    sonata.media.provider.vimeo:
+        class: Sonata\MediaBundle\Provider\VimeoProvider
+
+        arguments:
+            - sonata.media.provider.vimeo
+            - '@sonata.media.filesystem.local'
+            - '@sonata.media.cdn.server'
+            - '@sonata.media.generator.default'
+            - '@sonata.media.thumbnail.format'
+            - '@sonata.media.buzz.browser'
+            - '@sonata.media.metadata.proxy'
+
+        calls:
+            -
+                - setTemplates
+                - - helper_thumbnail: '@@SonataMedia/Provider/thumbnail.html.twig'
+                    helper_view: '@@SonataMedia/Provider/view_vimeo.html.twig'
```

Comment:
Regardless the result is the same, maybe the "expanded" syntax is easier to understand. IMO, the explicit indentation provides a clearer example about the nodes and hierarchy. 
```suggestion
                - 
                    -
                        helper_thumbnail: '@@SonataMedia/Provider/thumbnail.html.twig'
                        helper_view: '@@SonataMedia/Provider/view_vimeo.html.twig'
```

Reply:
You should decide with @greg0ire 😄
See https://github.com/sonata-project/SonataMediaBundle/pull/1748#discussion_r436097655

Reply:
I'm fine with either, the one that is not comfortable with that is OP (many devs have a phobia of linebreaks for some reason :stuck_out_tongue: )

Reply:
Every new line should be payed to someone 😆

Reply:
I didnt understand, is the current code acceptable or there is a need to change something?

Reply:
It's acceptable to me, but not to @phansys

Reply:
That was a suggestion. I don't consider this as a blocker. If you agree this way we are fine, just dismiss my suggestion :+1: